### ISSUE TITLE
Doc fails on raw string literals

### ISSUE THREAD

**Issue by ManuelSchneid3r** [doxygen/doxygen#11166]
**Describe the bug**
```
QRegularExpression separator_regex = QRegularExpression(R"([\s\\\/\-\[\](){}#!?<>"'=+*.:,;_]+)");
```
This makes the containing class not being documented.
Even worse it does not even throw an error.
Removing the raw string literal `R"(…)"` makes the class appear.

**Comment by albert-github** [comment]
I don't think this is a problem in doxygen but a problem in your raw string there is a double quote (`"`) and this will terminate the raw string.I think the `"` should be replaced by `"\"`.

Furthermore always:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).
- Please also specify the **full** doxygen version used (`doxygen -v`).

**Comment by ManuelSchneid3r** [comment]
It's the sole point of raw strings to not have to escape things. Please. Sure I could make a zip of an example but you could also add this line to any project you have locally. I strongly assume this is not about qregularexpression so just drop it an use a std::string.

**Comment by albert-github** [comment]
> It's the sole point of raw strings to not have to escape things.
>  I strongly assume this is not about qregularexpression so just drop it an use a std::string.

That is true and I think at other places this is working, e.g. defining just a raw string variable and giving this to the constructor.

> Sure I could make a zip of an example but you could also add this line to any project you have locally.

It is very easy for developers to oversee some small things that might lie hidden somewhere in the specific user code (also in the settings file!). Therefor it is necessary to have the example from the user and also know the **full** version of doxygen used by the user.

**Comment by ManuelSchneid3r** [comment]
Sure. doxygen: stable 1.12.0 (bottled), HEAD from homebrew. There you go:

[doxygen_example.zip](https://github.com/user-attachments/files/17226824/doxygen_example.zip)

**Comment by albert-github** [comment]
I had a solution for the initial example, but the example in https://github.com/doxygen/doxygen/issues/11166#issuecomment-2387977680 gave with that solutions some problems. Extended fix.

I've just pushed a proposed patch, pull request #11167

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
